### ISSUE TITLE
[BUG] Can't rename charts & other chart issues

### ISSUE THREAD

**Issue by chrisafl** [mage-ai/mage-ai#4691]
### Mage version

v0.9.65

### Describe the bug

All of these issues were produced using the demo.mage.ai website, on the exciting_wildflower pipeline.

https://demo.mage.ai/pipelines/exciting_wildflower/dashboard

I believe there are a few bugs here.
1) When you click "Create new chart" it seems to automatically template it off of the last chart that was modified.
2) If you have recently modified a chart and click new chart, the name of the new chart will be your existing chart and it seems that any change you make to it will override the existing chart
3) At no point in the process am I able to rename the chart
4) Sometimes when I click "Remove Chart" while testing, it also pops up the last chart I edited into edit mode

### To reproduce

1) Create a chart, then create a second chart
2) Edit a chart, save changes, then create a new chart
3) Create a new chart, or edit an existing chart and attempt to change the name
4) This one was intermittent, just keep creating, modifying, and deleting charts and it will happen

### Operating system

Windows 11, Edge Browser

**Comment by chrisafl** [comment]
On the table chart it shows row numbers, and that counter starts at 0. In the context of row numbers I think it makes more sense for counting to start at 1.

**Comment by chrisafl** [comment]
I am not able to get "Custom code for chart" to work at all. I built the Titanic pipeline example step by step, working well.

I am attempting to make a table chart on the dashboard using the custom code from the documentation, to filter the results of the dashboard to only those who are 21 and over:
https://docs.mage.ai/visualizations/dashboards

```python
from typing import List


@data_source
def filter_data(block_data_ouput: List, **kwargs) -> List:
    df_partition_1, df_partition_2 = block_data_ouput

    df_1 = df_partition_1[df_partition_1['age'] >= 21]
    df_2 = df_partition_2[df_partition_2['age'] >= 21]

    return [df_1, df_2]
```

When I attempt this the column count spinner goes forever and the table does not work.

**Comment by johnson-mage** [comment]
> I am not able to get "Custom code for chart" to work at all. I built the Titanic pipeline example step by step, working well.
> 
> I am attempting to make a table chart on the dashboard using the custom code from the documentation, to filter the results of the dashboard to only those who are 21 and over: https://docs.mage.ai/visualizations/dashboards
> 
> ```python
> from typing import List
> 
> 
> @data_source
> def filter_data(block_data_ouput: List, **kwargs) -> List:
>     df_partition_1, df_partition_2 = block_data_ouput
> 
>     df_1 = df_partition_1[df_partition_1['age'] >= 21]
>     df_2 = df_partition_2[df_partition_2['age'] >= 21]
> 
>     return [df_1, df_2]
> ```
> 
> When I attempt this the column count spinner goes forever and the table does not work.

@chrisafl Made an update to the docs that might've been causing the issue. The "age" column should be uppercase without making any transformations to the original titanic dataset, so it should be `Age` instead of `age`. Here's a screenshot of my chart settings that loaded fine for me:
<img width="1534" alt="image" src="https://github.com/mage-ai/mage-ai/assets/78053898/7d2c93a9-2a8e-49ac-85f3-6185e2d63d8e">

My pipeline `delicate_dew` only had 1 block `titanic_loader2` with these contents:
```
import io
import pandas as pd
import requests
from pandas import DataFrame

if 'data_loader' not in globals():
    from mage_ai.data_preparation.decorators import data_loader


@data_loader
def load_data_from_api(**kwargs) -> DataFrame:
    """
    Template for loading data from API
    """
    url = 'https://github.com/mage-ai/datasets/blob/master/titanic_survival.csv?raw=True'

    response = requests.get(url)
    return pd.read_csv(io.StringIO(response.text), sep=',')
```

**Comment by johnson-mage** [comment]
@chrisafl This PR https://github.com/mage-ai/mage-ai/pull/4714 should address the UI issues you mentioned based on my testing. Will be available in the next version of Mage (0.9.66).

**Comment by johnson-mage** [comment]
v0.9.66 was just released
